src/pygenesis/http_helper.py

```diff
@@ -74,14 +75,220 @@ def get_data_from_endpoint(endpoint: str, method: str, params: dict) -> str:
+                    endpoint=endpoint, method=method, params=new_params
+                )
+                # jobs_response = load_data(endpoint=endpoint, method=method, params=new_params, as_json=True)
+                jobs_catalogue_params, job_id = _jobs_job_id(
+                    jobs_response, params
+                )
+                response = json.loads(
+                    _jobs_catalogue_process(jobs_catalogue_params, job_id)
+                )
+                print(response)
+    except json.decoder.JSONDecodeError:
+        pass
+
+    response.encoding = "UTF-8"
+    print(response.json())
     _check_invalid_status_code(response.status_code)
     _check_invalid_destatis_status_code(response)
 
     return str(response.text)
 
 
+# TODO: test (Marco)
+def _generic_status_dict(
+    status_code: int, status_content: str, status_type: str
+) -> requests.Response:
+    """
+    Helper method which creates generic status dict.
+
+    Args:
+        status_code (int): Status code of the response
+        status_content (str): Status content of the response
+        status_type (str): Status type of the response (Information/ Warnung/ Fehler)
+
+    Returns:
+        requests.Response: the response from Destatis
+    """
+    # set up personalized requests.Reponse
+    request_status = requests.Response()
+    request_status.status_code = 42
+
+    # define status warning and texts
+    status_dict = {
+        "Status": {
+            "Code": status_code,
+            "Content": status_content,
+            "Type": status_type,
+        },
+    }
+
+    request_status._content = json.dumps(status_dict).encode("utf-8")
+
+    return request_status
+
+
+def _jobs_params(params: dict) -> dict:
+    """
+    Helper method which handles too large data requests with option of starting a job.
+
+    Args:
+        params (dict): dictionary of query parameters
+
+    Returns:
+        dict: new dict to start a job
+    """
+    # matching cases for user inputs
+    positive = ["ja", "j", "y", "yes"]
+    negative = ["nein", "n", "no"]
+
+    job_bool = ""
+    while job_bool.lower() not in positive + negative:
+        # get user input whether to start a job
+        logger.warning(
+            "Die Daten sind zu groß um direkt abgerufen zu werden."
+            + "Es muss ein Job gestartet werden, der einige Sekunden braucht,"
+            + "um die Daten bereitzustellen und abzurufen."
+            + "\n Sollen wir einen Job starten?"
+            + "\n Ja/Nein:"
+        )
+        job_bool = input("Sollen wir einen Job starten? \n Ja/Nein")
+        if job_bool.lower() not in (positive + negative):
+            logger.warning(
+                "Keinen Input erhalten, es wird kein Job angestoßen."
+            )
+            job_bool = "nein"
+
+    # retry request with job parameter set to True
+    if job_bool.lower() in positive:
+        params |= {"job": "true"}
+
+    else:
+        params = None
```

Comment:
Warum soll das params dict hier gelöscht werden? Sollte man das übergebene dict nicht unverändert lassen?

Reply:
Das Problem ist hier dass dein return value dann auc hnicht stimmt, weil die Funktion kein dict mehr zurückgibt. Ich würde vorschlagen, eher "job" auf "false" zu setzen und das in der parent function abzufragen, also nicht auf `isinstance(job_params, dict)` prüfen sondern auf `if job_params.get("job") == "true"`, denn wir wollen ja nur den Job anstoßen, wenn der Wert auf "true" gesetzt wurde